### PR TITLE
Add Support for Symfony 6

### PR DESCRIPTION
**Pull request by TheMilek**
continuation of / to potential merge after merging https://github.com/Sylius/ShopApiPlugin/pull/734

### CODE REVIEW

**Review by loic425 — APPROVED** [review]

**Review by jakubtobiasz — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -56,6 +56,9 @@
     "extra": {
         "branch-alias": {
             "dev-master": "1.0-dev"
+        },
+        "symfony": {
+            "require": "^6.0"
```
Comment (coldic3) [inline composer.json]:
```suggestion
            "require": "^5.4 || ^6.0"
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -18,17 +18,22 @@ jobs:
         strategy:
             fail-fast: false
             matrix:
-                php: [8.0, 8.1]
-                node: [10.x]
-                mysql: [5.7, 8.0]
-                symfony: [^4.4, ~5.2.0, ~5.3.0]
```
Comment (coldic3) [inline .github/workflows/build.yml]: I think I told you about it already that it is IMO a good idea to drop SF 4.4 in a separate PR. But if you want to keep it as it is, let's add it at least to the PR title to have this info in the generated changelog on next release :grinning:

Comment (TheMilek) [reply .github/workflows/build.yml]: This PR is continuation of https://github.com/Sylius/ShopApiPlugin/pull/734
I'm planning to make it in the order the above one [this one](https://github.com/Sylius/ShopApiPlugin/pull/735) after rebase and as a last one to merge will be this support to SF6 as it's based on the Sylius 1.12 support, so the changelog will be informative :dancer:

File: tests/Responses/Expected/address_book/SF6/validation_create_address_book_with_wrong_province_response.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+      "code": 400,
```
Comment (coldic3) [inline tests/Responses/Expected/address_book/SF6/validation_create_address_book_with_wrong_province_response.json]:
```suggestion
  "code": 400,
```

File: tests/Responses/Expected/address_book/SF6/validation_create_address_book_with_wrong_country_response.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+            "countryCode": {
+                "errors": [
+                    "The selected choice is invalid."
+                ]
+            },
+            "street": [],
+            "city": [],
+            "postcode": []
+        }
+}
```
Comment (coldic3) [inline tests/Responses/Expected/address_book/SF6/validation_create_address_book_with_wrong_country_response.json]:
```suggestion
    }
```

File: tests/Controller/AddressBook/CreateAddressApiTest.php [hunk 5]
```diff
@@ -110,7 +111,13 @@ public function it_does_not_allow_user_to_add_new_address_to_address_book_withou
...
 
         $response = $this->createAddress($data);
-        $this->assertResponse($response, 'address_book/validation_create_address_book_with_wrong_country_response', Response::HTTP_BAD_REQUEST);
+        if (Kernel::VERSION_ID < 60000) {
+            $this->assertResponse($response, 'address_book/SF5/validation_create_address_book_with_wrong_country_response', Response::HTTP_BAD_REQUEST);
+        }
+
+        if (60000 <= Kernel::VERSION_ID) {
+            $this->assertResponse($response, 'address_book/SF6/validation_create_address_book_with_wrong_country_response', Response::HTTP_BAD_REQUEST);
+        }
```
Comment (coldic3) [inline tests/Controller/AddressBook/CreateAddressApiTest.php]:
```suggestion
        if (Kernel::VERSION_ID < 60000) {
            $this->assertResponse($response, 'address_book/SF5/validation_create_address_book_with_wrong_country_response', Response::HTTP_BAD_REQUEST);
        } else {
            $this->assertResponse($response, 'address_book/SF6/validation_create_address_book_with_wrong_country_response', Response::HTTP_BAD_REQUEST);
        }
```

or with an early return.

File: tests/Controller/AddressBook/CreateAddressApiTest.php [hunk 6]
```diff
@@ -135,7 +142,13 @@ public function it_does_not_allow_user_to_add_new_address_to_address_book_withou
...
 
         $response = $this->createAddress($data);
-        $this->assertResponse($response, 'address_book/validation_create_address_book_with_wrong_province_response', Response::HTTP_BAD_REQUEST);
+        if (Kernel::VERSION_ID < 60000) {
+            $this->assertResponse($response, 'address_book/SF5/validation_create_address_book_with_wrong_province_response', Response::HTTP_BAD_REQUEST);
+        }
+
+        if (60000 <= Kernel::VERSION_ID) {
+            $this->assertResponse($response, 'address_book/SF6/validation_create_address_book_with_wrong_province_response', Response::HTTP_BAD_REQUEST);
+        }
```
Comment (coldic3) [inline tests/Controller/AddressBook/CreateAddressApiTest.php]:
```suggestion
        if (Kernel::VERSION_ID < 60000) {
            $this->assertResponse($response, 'address_book/SF5/validation_create_address_book_with_wrong_province_response', Response::HTTP_BAD_REQUEST);
        } else {
            $this->assertResponse($response, 'address_book/SF6/validation_create_address_book_with_wrong_province_response', Response::HTTP_BAD_REQUEST);
        }
```

or with an early return.

**Review by coldic3 — APPROVED** [review]

**Review by coldic3 — APPROVED** [review]

File: tests/Controller/AddressBook/CreateAddressApiTest.php [hunk 7]
```diff
@@ -110,7 +111,11 @@ public function it_does_not_allow_user_to_add_new_address_to_address_book_withou
 JSON;
 
         $response = $this->createAddress($data);
-        $this->assertResponse($response, 'address_book/validation_create_address_book_with_wrong_country_response', Response::HTTP_BAD_REQUEST);
+        if (Kernel::VERSION_ID < 60000) {
+            $this->assertResponse($response, 'address_book/SF5/validation_create_address_book_with_wrong_country_response', Response::HTTP_BAD_REQUEST);
+        } else {
+            $this->assertResponse($response, 'address_book/SF6/validation_create_address_book_with_wrong_country_response', Response::HTTP_BAD_REQUEST);
+        }
```
Comment (GSadee) [inline tests/Controller/AddressBook/CreateAddressApiTest.php:118]: To avoid that, we could probably define the custom validation message

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Kamil! :tada:

**Pull request merged**